File: 2023/day01/Python/part1.py
import re


inp = open('../input.txt', 'r').readlines()

calib_sum = 0


for line in inp[:-1]:
    nums = re.findall('\d', line)

    calibration_value = nums[0] + nums[-1]
    calib_sum += int(calibration_value)


print(calib_sum)

File: 2023/day01/Lua/part1.lua
local lines = io.open('../input.txt', 'r'):lines()

local calib_sum = 0


for line in lines do
    if line ~= '' then
        local first = string.match(line,           '%d')
        local last  = string.match(line:reverse(), '%d')

        calib_sum = calib_sum + (first .. last)
    end
end


io.write(calib_sum)